src/backend/internal/controller/user/get_user_info.go

```diff
@@ -0,0 +1,42 @@
+package user
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/Roongkun/software-eng-ii/internal/model"
+	"github.com/gin-gonic/gin"
+)
+
+func (r *Resolver) GetUserInfo(c *gin.Context) {
+	UserIdInput := model.UserIdInput{}
+	// Parse JSON body to extract user ID
+	if err := c.BindJSON(&UserIdInput); err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"error": "Failed to parse request body"})
```

Comment:
Please consider adding `c.Abort()` before returning. 
This will tell `gin` to abort the request immediately after finishing this handler.

Actually, this may not be necessary at the moment as this handler serves as an endpoint. However, from my pov, this handler could potentially become a middleware in the future. Hence, adding it now would not be a bad choice.